packages/@cdklabs/cdk-cicd-wrapper-cli/src/cmds/LicenseCommand.ts

```diff
@@ -145,719 +128,113 @@ class LicenseChecker {
-    context: ScanningContext,
-    pythonProjectToCheck: string,
-    venvLocation: string,
-    tmpNoticeLocation: string,
-  ) {
-    const commandArgs = [
-      // Suppressed as no user input it used to manage the path and child_process
-      // nosemgrep
-      `--python=${path.join(venvLocation, 'bin', 'python')}`,
-      '--format=plain-vertical',
-      '--with-license-file',
-      '--no-license-path',
-    ];
+  private runLicenseCheckers(project: Projects) {
+    logger.debug('Scanning projects ...');
 
-    this.addPythonPackageExclusion(commandArgs);
-
-    const options: SpawnSyncOptions = {
-      encoding: 'utf8',
-      stdio: ['ignore', openSync(tmpNoticeLocation, 'w'), 0],
-      ...GLOBAL_TIMEOUT,
-    };
-
-    const commandResults = spawnSync(
-      // Suppressed as no user input it used to manage the path and child_process
-      // nosemgrep
-      path.join(context.workingDir, '.venv', 'bin', PYTHON_LICENSE_CHECKER_TOOL),
-      commandArgs,
-      options,
-    );
-
-    if (commandResults.status !== 0) {
-      console.error(`Module ${pythonProjectToCheck} failed the license check.`);
-      throw new Error(`Module ${pythonProjectToCheck} failed the license check.`);
-    }
-  }
-
-  private createScanningEnvironment(scanning: (context: ScanningContext) => number) {
-    let workingDir;
-    let exitCode = 0;
-    try {
-      if (this.debug) {
-        workingDir = path.join(process.cwd(), 'license-debug');
-
-        if (existsSync(workingDir)) {
-          rmSync(workingDir, { recursive: true });
-        }
-
-        mkdirSync(workingDir);
-      } else {
-        workingDir = mkdtempSync(path.join(os.tmpdir(), 'license'));
-      }
-
-      const venvLocation = path.join(workingDir, '.venv');
-
-      const pythonCommands = CliHelpers.getPythonCommand();
-      // Suppressed as no user input it used to manage the path and child_process
-      // nosemgrep
-      const vEnvCreation = spawnSync(pythonCommands.pythonExecutable, ['-m', 'venv', venvLocation]);
-
-      if (vEnvCreation.status !== 0) {
-        throw new Error(`Failed to create virtual environment ${venvLocation}`);
-      }
-
-      exitCode = scanning({
-        projectRoot: process.cwd(),
-        workingDir: workingDir,
-        // Suppressed as no user input it used to manage the path and child_process
-        // nosemgrep
-        pip: path.join(venvLocation, 'bin', pythonCommands.pipExecutable),
-        // Suppressed as no user input it used to manage the path and child_process
-        // nosemgrep
-        python: path.join(venvLocation, 'bin', pythonCommands.pythonExecutable),
-      });
-    } catch (error) {
-      console.error(error);
-      console.error('License scan failed');
-      exitCode = 1;
-    } finally {
-      if (workingDir && !this.debug) {
-        rmSync(workingDir, { recursive: true });
-      }
-    }
-
-    return exitCode;
+    Object.entries(COLLECTORS).map(([type, collectorCtor]) =>
```

Comment:
Just want to double check with you if 'collectorCtor' is a typo or an intended name

Reply:
Ctor is a shorthand for constructor